From 6e543f39f6bec6eb2e02eea02029c7f4ec533b66 Mon Sep 17 00:00:00 2001
From: Johannes Pointner <johannes.pointner@br-automation.com>
Date: Fri, 1 Dec 2023 11:02:39 +0100
Subject: [PATCH] ldconfig: add usrmerge support

Check whether SLIBDIR is a symlink, which is the case if usrmerge
is enabled, and if so, ignore it.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Johannes Pointner <johannes.pointner@br-automation.com>
---
 ldconfig.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ldconfig.c b/ldconfig.c
index e826410..72ac67b 100644
--- a/ldconfig.c
+++ b/ldconfig.c
@@ -1371,10 +1371,16 @@ main (int argc, char **argv)
 
   if (!opt_only_cline)
     {
+      struct stat buf;
+      int ret;
       parse_conf (config_file, true);
 
       /* Always add the standard search paths.  */
-      add_system_dir (SLIBDIR);
+      /* Check whether SLIBDIR is a symlink, which is the case if usrmerge
+       is enabled, and if so, ignore it. */
+      ret = lstat(SLIBDIR ,&buf);
+      if(ret == -1 || !S_ISLNK(buf.st_mode))
+        add_system_dir (SLIBDIR);
       if (strcmp (SLIBDIR, LIBDIR))
 	add_system_dir (LIBDIR);
       add_system_dir (SLIBDIR32);
